### DISCUSSION TITLE
Programmatically spin up Clerk apps

### DISCUSSION THREAD

**Discussion by pedromiranda96** [clerk/javascript#2279] [Feature Requests]
My team and I are working on a product that relies on Clerk to handle user authentication. Our idea is for each customer to have their own app instance and we thought we should automate that process. We started building this feature into the app users access to get the product and we were able to deploy a PlanetScale database as well as a Netlify website via REST API calls, but so far we have not been able to find a way to programmatically spin up a Clerk App. I know Vercel also has a REST API for creating deployments among other things, so we thought maybe that's not so far-fetched.

I apologize if that has already been discussed or maybe if it just doesn't make sense for Clerk, but I wonder if someone has already tried to achieve something like that (and succeeded, maybe we haven't looked at the right places) or if it's even something that the Clerk team would consider adding to their roadmap. 

EDIT:
Auth0 appears to have something like that: https://auth0.com/docs/api/management/v2/clients/post-clients

**Comment by LekoArts** [comment]
Hi, please upvote and leave your comment on https://feedback.clerk.com/roadmap/022e3bb0-31c2-4c29-a39f-9fc77480394f - thanks!
